### PR TITLE
[ie/xvideos] Fix regex (#9493)

### LINKED ISSUE #9493: [xvideos] not supported new url format
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

BR

### Provide a description that is worded well enough to be understood

new xv url format is not compatible anymore to yt-dlp, download only low quality

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
# yt-dlp --hls-prefer-native -o video.mp4 'https://www.xvideos.com/video.uckofkv4e3d/teste_de_madrasta_e_meia-irma_se_sou_gay_-_macy_meadows_e_sydney_paige_-_terapia_familiar_-_alex_adams' -vU
[debug] Command-line config: ['--hls-prefer-native', '-o', 'video.mp4', 'https://www.xvideos.com/video.uckofkv4e3d/teste_de_madrasta_e_meia-irma_se_sou_gay_-_macy_meadows_e_sydney_paige_-_terapia_familiar_-_alex_adams', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.03.10 from yt-dlp/yt-dlp [615a84447] (zip)
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-5.10.0-23-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1w  11 Sep 2023, glibc 2.31)
[debug] exe versions: ffmpeg 5.1.1-static (setts), ffprobe 5.1.1-static
[debug] Optional libraries: certifi-2024.02.02, requests-2.31.0, sqlite3-3.34.1, urllib3-2.2.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1803 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.03.10 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.03.10 from yt-dlp/yt-dlp)
[generic] Extracting URL: https://www.xvideos.com/video.uckofkv4e3d/teste_de_madrasta_e_meia-irma_se_sou_gay_-_macy_meadows_e_sydney_paige_-_terapia_familiar_-_alex_adams
[generic] teste_de_madrasta_e_meia-irma_se_sou_gay_-_macy_meadows_e_sydney_paige_-_terapia_familiar_-_alex_adams: Downloading webpage
[redirect] Following redirect to https://www.xvideos.com/video.uckofkv4e3d/step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams
[generic] Extracting URL: https://www.xvideos.com/video.uckofkv4e3d/step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams
[generic] step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams: Extracting information
[debug] Looking for embeds
[debug] Identified a JSON LD
[generic] Extracting URL: https://cdn77-vid-mp4.xvideos-cdn.com/heDEl7YzaARYDM6vhAfohA==,1710857035/videos/mp4/1/2/d/xvideos.com_12d8dd3274f9ccfd1246bbe532f238b1.mp4?ui=MTM1LjE4MS42Mi4xNDItL3ZpZGVvLnVja29ma3Y0ZTNkL3N0ZXBfbW9tX2Fu#__youtubedl_smuggle=%7B%22force_videoid%22%3A+%22step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams%22%2C+%22to_generic%22%3A+true%2C+%22referer%22%3A+%22https%3A%2F%2Fwww.xvideos.com%2Fvideo.uckofkv4e3d%2Fstep_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams%22%7D
[generic] step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams: Downloading webpage
[debug] Identified a direct video link
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] step_mom_and_step_sister_test_if_i_m_gay_-_macy_meadows_and_sydney_paige_-_family_therapy_-_alex_adams: Downloading 1 format(s): mp4
[debug] Invoking http downloader on "https://cdn77-vid-mp4.xvideos-cdn.com/heDEl7YzaARYDM6vhAfohA==,1710857035/videos/mp4/1/2/d/xvideos.com_12d8dd3274f9ccfd1246bbe532f238b1.mp4?ui=MTM1LjE4MS42Mi4xNDItL3ZpZGVvLnVja29ma3Y0ZTNkL3N0ZXBfbW9tX2Fu"
[download] Destination: video.mp4
[download] 100% of   43.59MiB in 00:00:00 at 82.71MiB/s
```

### PR DESCRIPTION
**Pull request by sta1us**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Add support new url format

Fixes #9493

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by seproDev** [comment]
Please fill out the PR template and add tests.

**Comment by sta1us** [comment]
@seproDev  is that correct? =)

File: yt_dlp/extractor/xvideos.py [hunk 1]
```diff
@@ -90,6 +90,51 @@ class XVideosIE(InfoExtractor):
...
+    }, {
+        'url': 'http://static-hw.xvideos.com/swf/xv-player.swf?id_video=ucuvbkfda4e',
+        'only_matching': True,
+    }, {
+        'url': 'http://xvideos.com/video.ucuvbkfda4e/a_beautiful_red-haired_stranger_was_refused_but_still_came_to_my_room_for_sex',
+        'only_matching': True
+    }, {
+        'url': 'https://xvideos.com/video.ucuvbkfda4e/a_beautiful_red-haired_stranger_was_refused_but_still_came_to_my_room_for_sex',
+        'only_matching': True
+    }, {
```
Comment (bashonly) [inline yt_dlp/extractor/xvideos.py:108]: these 2 are dupes, can remove

File: yt_dlp/extractor/xvideos.py [hunk 2]
```diff
@@ -90,6 +90,51 @@ class XVideosIE(InfoExtractor):
...
+        'url': 'https://it.xvideos.com/video.ucuvbkfda4e/a_beautiful_red-haired_stranger_was_refused_but_still_came_to_my_room_for_sex',
+        'only_matching': True
+    }, {
+        'url': 'http://de.xvideos.com/video.ucuvbkfda4e/a_beautiful_red-haired_stranger_was_refused_but_still_came_to_my_room_for_sex',
+        'only_matching': True
+    }, {
+        'url': 'https://de.xvideos.com/video.ucuvbkfda4e/a_beautiful_red-haired_stranger_was_refused_but_still_came_to_my_room_for_sex',
+        'only_matching': True
```
Comment (bashonly) [inline yt_dlp/extractor/xvideos.py:137]: we really only need an .es test out of these

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/xvideos.py (hunk 1 above)
Comment (sta1us) [reply yt_dlp/extractor/xvideos.py:108]: these are not duplicates. http/http**s**

File: yt_dlp/extractor/xvideos.py (hunk 2 above)
Comment (sta1us) [reply yt_dlp/extractor/xvideos.py:137]: Just copied the previous test. If the others are not needed, I can delete them.

File: yt_dlp/extractor/xvideos.py (hunk 1 above)
Comment (bashonly) [reply yt_dlp/extractor/xvideos.py:108]: the first test highlighted here is an exact dupe of the very first test in the `_TESTS` list, no?

and we don't need to test http vs https, it is convention to always match both http and https in the `_VALID_URL`

Comment (sta1us) [reply yt_dlp/extractor/xvideos.py]: i remove 103-108 lines.

File: yt_dlp/extractor/xvideos.py (hunk 2 above)
Comment (sta1us) [reply yt_dlp/extractor/xvideos.py]: remove the others, except .es

**Review by bashonly — APPROVED** [review]

**Comment by bashonly** [comment]
> ### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
> - [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
> - [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://> unlicense.org/) (provide reliable evidence)

who is?

**Comment by sta1us** [comment]
@bashonly I understand that if you wrote this extractor, then you are the original author. If you just fix it, the you are not original author. Is that correct? 
If this is not correct then I am the original author of this code.

**Comment by Grub4K** [comment]
No, its about the added code. The old code is already licensed under "Unlicense". The new code may not be which is why we ask you to explicitly state you license it as such. If there are mutliple authors that wrote the code, or you took the code from another Unlicense source, then the second checkbox comes into play

**Comment by sta1us** [comment]
@Grub4K ок, i understand. thanks

upd.fixed

**Comment by WanHaes** [comment]
What should I do to fix downloading, is it possible now? I put the edited xvideos.py in "extractor" folder and it didn't work

**Comment by sta1us** [comment]
@WanHaes 
1. run devscripts/lazy_load_template.py
2. copy extractor/lazy_extractors.py to extractor folder 

or just edit lazy_extractors.py if file exists.
```python
class XVideosIE(LazyLoadExtractor):
    _module = 'yt_dlp.extractor.xvideos'
    IE_NAME = 'XVideos'
    _VALID_URL = '(?x)\n                    https?://\n                        (?:\n                            (?:[^/]+\\.)?xvideos2?\\.com/video\\.?|\n                            (?:www\\.)?xvideos\\.es/video\\.?|\n                            (?:www|flashservice)\\.xvideos\\.com/embedframe/|\n                            static-hw\\.xvideos\\.com/swf/xv-player\\.swf\\?.*?\\bid_video=\n                        )\n                        (?P<id>[0-9a-z]+)\n                    '
    age_limit = 18
    _RETURN_TYPE = 'video'
```

**Comment by WanHaes** [comment]
@bashonly @sta1us 
Thanks. I use Tartube which runs in the MSYS2 environment on Windows. By default Tartube updates the yt-dlp with `--no-dependencies` option. 

Using `../../../mingw64/bin/pip3.exe install --no-deps -U "yt-dlp @ https://github.com/yt-dlp/yt-dlp/archive/master.tar.gz"` ignores no-deps and installation fails. So I installed fresh yt-dlp with pip, upgraded it to master and just copied whole yt_dlp folder to Tartube's location. And now the downloads are working properly, yay.

**Pull request merged**